app/src/org/commcare/android/analytics/TimedStatsTracker.java

```diff
@@ -0,0 +1,109 @@
+package org.commcare.android.analytics;
+
+import android.content.SharedPreferences;
+import android.util.Log;
+
+import org.commcare.dalvik.application.CommCareApplication;
+
+import java.util.Date;
+
+/**
+ * Utils for keeping track of the length of some app events, and then reporting them to Google
+ * analytics
+ *
+ * @author amstone
+ */
+public class TimedStatsTracker {
+
+    private static final String TAG = TimedStatsTracker.class.getSimpleName();
+
+    private static String KEY_LAST_FORM_NAME = "last-form-name";
```

Comment:
I'm not 100% sure I think it's clean for us to record this. It's app/user specific data which makes it feel like it shouldn't bubble up to the analytics framework. It's also fairly tied towards a user's specific behavior, so if we're assuming this isn't "secure" data, it sketches me out that we'd be correlating position, time , and a form, that feels a lot like a patient identifying event.